### ISSUE TITLE
Suggestion: Redundant _ variables in structs

### ISSUE THREAD

**Issue by Wilfred** [Manishearth/rust-clippy#479]
Given the code:

```rust
struct Foo {
    a: i64,
    b: i64,
    c: i64,
    d: i64,
}

fn use_foo(f: Foo) {
    match f {
        Foo { a: _, b: b_var, .. } => {
            println!("b_var is {:?}", b_var);
        }
    }
}
```

The mistake here is that the user thinks they need to match `a` before matching `b`. This function could be written:

```rust
fn use_foo(f: Foo) {
    match f {
        Foo { b: b_var, .. } => {
            println!("b_var is {:?}", b_var);
        }
    }
}
```

I understand if you think this is too specific a use case, but I have encountered this in some of my own code

**Comment by Manishearth** [comment]
Nah, sounds like a good idea. Pretty easy to implement, it should be an EarlyPass checking for struct patterns with `_`s

**Comment by GuillaumeGomez** [comment]
I'll give it a try.

**Comment by untitaker** [comment]
Can be closed due to #482

**Comment by GuillaumeGomez** [comment]
Not yet, it's not complete.

**Comment by GuillaumeGomez** [comment]
Closed by #535.
